src/extensions/surveys.tsx

```diff
@@ -483,6 +485,70 @@ export function SurveyPopup({
     const shouldShowConfirmation = isSurveySent || previewPageIndex === survey.questions.length
     const confirmationBoxLeftStyle = style?.left && isNumber(style?.left) ? { left: style.left - 40 } : {}
 
+    // Add URL change listener to hide survey when URL no longer matches
+    useEffect(() => {
+        if (isPreviewMode || !survey.conditions?.url || !posthog) {
+            return
+        }
+
+        const checkUrlMatch = () => {
+            const urlMatchType = survey.conditions?.urlMatchType ?? 'icontains'
+            const url = survey.conditions?.url
+            if (!url) {
+                return
+            }
+            const urlCheck = surveyUrlValidationMap[urlMatchType](url)
+            if (!urlCheck) {
+                setIsPopupVisible(false)
+                removeSurveyFromFocus(survey.id)
+            }
+        }
+
+        // Listen for navigation events
+        const handleUrlChange = () => {
+            checkUrlMatch()
+        }
+
+        // Listen for browser back/forward
+        window.addEventListener('popstate', handleUrlChange)
+
+        // Create our before_send handler
+        const surveyBeforeSend = (eventData: CaptureResult | null) => {
+            if (eventData?.event === '$pageview' || eventData?.event === '$pageleave') {
```

Comment:
if autocapture is disabled, those events aren't fired, so this bug would still happen

Reply:
changed the approach